### PR TITLE
Configuration to allow newer smart strips (HS300 specifically)

### PR DESCRIPTION
**Pull request by germtuna**
I did not add additional functionality per se, only the ability to see and configure sockets on the HS300 (and presumably other >3 socket Kasa strips). 
Tested functionality running:
- python-kasa => 0.4.0dev3
- OctoPrint 1.7.0rc3
- Python 3.7.3
- OctoPi 0.18.0.0

 I did not regression test against other smartstrips (I only have the HS300).

### CODE REVIEW

**Comment by germtuna** [comment]
Nothing drastic, just support for the HS300.

**Comment by jamesmccannon02** [comment]
Thanks for the contribution, I'll take a look and then commit it into the main branch when I get a chance. J.

**Comment by germtuna** [comment]
No worries.  It occurred to be that I did not update the relevant bits in README.md. Let me know if you want me to do so.

File: octoprint_TpLinkAutoShutdown/__init__.py [hunk 1]
```diff
@@ -140,6 +160,9 @@ def get_settings_defaults(self):
 				deviceOne=True,
 				deviceTwo=True,
 				deviceThree=True,
+    			deviceFour=True,
```
Comment (jamesmccannon02) [inline octoprint_TpLinkAutoShutdown/__init__.py:163]: Don't know if this is just the way I am viewing the code on GitHub but this looks to be indented incorrectly. Could you please have a look as deviceFour to me looks as if it needs to be indented again to conform the the rules.

File: octoprint_TpLinkAutoShutdown/__init__.py [hunk 2]
```diff
@@ -167,6 +190,27 @@ def get_settings_defaults(self):
 					auto=False,
 					movieDone=False,
 				),
+    			deviceFour=dict(
```
Comment (jamesmccannon02) [inline octoprint_TpLinkAutoShutdown/__init__.py:193]: Hi again here could you please check your indentation here. This needs to be indented once more for the deviceFour dict. Thanks

**Comment by jamesmccannon02** [comment]
Thanks for that fix. Just having a loot at the other stuff. Looks good. Just out of interest have you tested it with the KP303 as well to ensure that it doesn't break anything?

File: octoprint_TpLinkAutoShutdown/__init__.py (hunk 2 above)
Comment (germtuna) [reply octoprint_TpLinkAutoShutdown/__init__.py]: Yeah not sure how that indent got borked. Fixed.

**Comment by jamesmccannon02** [comment]
If it all works and you were interested, the other thing that could be developed on top is a way to auto detect the type of device and then change the number of sockets shown on the UI to only show either 3 or 6 sockets dependent on the users device. This is possible through using the control lib that I am using as an interface to the sockets. Although I haven't done this myself as I don't understand the complex networking setup that OctoPrint uses when sending data too and from the website UI. Then again I'm not one for web development anyway.

**Comment by germtuna** [comment]
I only have the HS300 (6 outlets) so I could not test. Based on the behavior before the changes, it only detected the first 3 outlets on the HS300 but worked with the 3 it found.  I suspect (not looked) that the kasa python library just enumerates the device and passes it. That would lead me to assume that the KP303 would still work, but to your point the gui may not reflect the number of sockets appropriately. I will poke around a bit.

**Comment by jamesmccannon02** [comment]
sure. Ive made a couple of changes and will push them in a second back into this section. make sure you pull them back down once I've pushed them up.

**Comment by jamesmccannon02** [comment]
Hi give it a pull now and see the changes I have made I think it'll all work but please test it thoroughly before you confirm it works. Each plug one at a time as this will ensure all settings are working correctly.

**Comment by germtuna** [comment]
Pulled, Uninstalled, Reinstalled. Each plug was tested individually on/off successfully as well as all at once. Looks ok to me.

**Comment by jamesmccannon02** [comment]
fantastic. All that is left then is for me to change the version numbering in a few places so that the auto-update feature on OctoPrint will work and not annoy users with a popup. After that I can release it to the public. Thanks for you input.

**Pull request merged**